### PR TITLE
Sidebar improvements

### LINKED ISSUE #353: Display page visibility on pages
The only way to find out whether a page has been made visible to a certain group (e.g. if a minutes has been "published" as in made available to all students) is to edit the page and look at the visibility there. 

Two options are feasible:

- Display visibility on the page itself. This could be a subheading "visibility" in the sidebar that contains a comma-separated lists of all groups that the page is visible for.

### PR DESCRIPTION
**Pull request by abc013**
Closes #353.

Furthermore, this PR streamlines the minute and information page sidebar layout (see second commit) and also adds empty date or editor handling.

**I was unable to test some of the changes, specifically those about the minute sidebar. Before approving, please check whether everything looks as expected.** I'm investigating why minute pages don't work (even on main) on my local setup.

### CODE REVIEW

**Comment by frcroth** [comment]
It would be helpful to have a list of changes that are supposed to work to test them (if automated tests are not an option).

File: myhpi/core/templates/core/sidebar.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{% load core_extras %}
+{% load i18n %}
+
+{% if perms.wagtail.edit_page %}
```
Comment (frcroth) [inline myhpi/core/templates/core/sidebar.html:4]: This does not work for me. Do you have documentation mentioning this?

File: myhpi/core/templates/core/minutes.html [hunk 2]
```diff
@@ -8,69 +8,69 @@
...
-            <h4>{% translate "Participants" %}</h4>
-            <ul>
-                {% for participant in page.participants.all %}
-                    <li>{{ participant.get_full_name }}</li>
-                {% endfor %}
-            </ul>
-            <h4>{% translate "Guests" %}</h4>
-            {% if page.guests %}
+            <aside class="side-panel border-accent">
+                <h1 class="toc-title">{% translate "Date" %}</h1>
```
Comment (frcroth) [inline myhpi/core/templates/core/minutes.html:18]: This class seems to be misnamed now. Maybe side-panel-title?

File: myhpi/core/templates/core/minutes.html [hunk 3]
```diff
@@ -8,69 +8,69 @@
...
-                        <a href="{{ prev.url }}">< {% translate "Previous minutes" %}</a>
-                    {%  endif %}
-                </div>
-                <div class="col-6">
-                    {%  next_minutes page as next %}
-                    {% if next %}
-                        <a href="{{ next.url }}"> {% translate "Next minutes" %} ></a>
-                    {%  endif %}
-                </div>
+        <div id="minutes-navigation" class="row d-print-none">
```
Comment (frcroth) [inline myhpi/core/templates/core/minutes.html:60]: Not part of this PR, but I'm not really happy with the styling here (Missing space, the arrows look very programmer-art, maybe introduce actual buttons?)
image

**Comment by jeriox** [comment]
additional idea: display a warning banner or some indication in the sidebar for pages that are a draft (similar to the exclamation mark in the list)

File: myhpi/core/templates/core/minutes.html (hunk 2 above)
Comment (abc013) [reply myhpi/core/templates/core/minutes.html]: Fixed.

File: myhpi/core/templates/core/minutes.html (hunk 3 above)
Comment (abc013) [reply myhpi/core/templates/core/minutes.html:60]: Me too. But first make an issue out of it?

File: myhpi/core/templates/core/sidebar.html (hunk 1 above)
Comment (abc013) [reply myhpi/core/templates/core/sidebar.html:4]: Sadly, no, but it works on my side(tm).

**Comment by abc013** [comment]
@jeriox good idea. I added it in [`e56ed7d`](https://github.com/fsr-de/myHPI/pull/383/commits/e56ed7d1477b3d52b6701a994d567b079ea3decd):
image

**Comment by jeriox** [comment]
@abc013 is the description of the PR still correct or could you test your changes locally in the meantime?

File: myhpi/core/templates/core/information_page.html [hunk 4]
```diff
@@ -15,30 +15,24 @@ <h1 class="page-title">
     </div>
     <div class="col-lg-3">
         <div class="side-panel-container">
+            {% if request.is_preview %}
+                <aside class="side-panel border-accent toc-container d-print-none">
+                    <h1 class="side-panel-warning">{% translate "⚠️ Preview ⚠️" %}</h1>
```
Comment (jeriox) [inline myhpi/core/templates/core/information_page.html:20]: why is this not part of core/sidebar.html?

File: myhpi/locale/de/LC_MESSAGES/django.po [hunk 5]
```diff
@@ -18,122 +18,126 @@ msgstr ""
...
-msgstr "[link fehlt]"
-
 #: core/templates/core/information_page.html:19
 msgid "Last edited"
 msgstr "Zuletzt bearbeitet"
 
-#: core/templates/core/information_page.html:22
+#: core/templates/core/information_page.html:21
+msgid "A long time ago"
+msgstr ""
```
Comment (jeriox) [inline myhpi/locale/de/LC_MESSAGES/django.po:75]: translation missing

File: myhpi/core/templates/core/minutes.html [hunk 6]
```diff
@@ -8,69 +8,74 @@
...
-            <p>{{ page.author.get_full_name }}</p>
-            <h4>{% translate "Participants" %}</h4>
-            <ul>
-                {% for participant in page.participants.all %}
-                    <li>{{ participant.get_full_name }}</li>
-                {% endfor %}
-            </ul>
-            <h4>{% translate "Guests" %}</h4>
-            {% if page.guests %}
+            {% if request.is_preview %}
```
Comment (jeriox) [inline myhpi/core/templates/core/minutes.html:17]: this labels previews, in the list we are labelling drafts. Is this intentional?

File: myhpi/core/templates/core/information_page.html (hunk 4 above)
Comment (abc013) [reply myhpi/core/templates/core/information_page.html:20]: This is shown above all other entries in the sidebar. However, `core/sidebar.html` is copied below some other entries, depending on page type.

**Comment by abc013** [comment]
> @abc013 is the description of the PR still correct or could you test your changes locally in the meantime?

I would say I was able to test most of it, but would be happy to have someone with a proper database looking over it. My test minute pages are quite empty. :D

File: myhpi/locale/de/LC_MESSAGES/django.po (hunk 5 above)
Comment (abc013) [reply myhpi/locale/de/LC_MESSAGES/django.po]: fixed. Thanks!

File: myhpi/core/templates/core/minutes.html (hunk 6 above)
Comment (abc013) [reply myhpi/core/templates/core/minutes.html]: You are right, updated.

Although I am wondering how they differ: Does `page.live` mean that it only shows this label if there is no version published?

File: myhpi/core/templates/core/information_page.html (hunk 4 above)
Comment (jeriox) [reply myhpi/core/templates/core/information_page.html:20]: but is it necessary to have this on top? I just don't like the duplication between the two templates

File: myhpi/core/templates/core/minutes.html (hunk 6 above)
Comment (jeriox) [reply myhpi/core/templates/core/minutes.html]: see https://docs.wagtail.org/en/stable/reference/pages/model_reference.html#wagtail.models.Page.live

after reading this, checking for `page.has_unpublished_changes` probably is what we want

**Comment by frcroth** [comment]
> I would say I was able to test most of it, but would be happy to have someone with a proper database looking over it. My test minute pages are quite empty. :D

With #444 merged, you should now be able to create somewhat longer minutes automatically.

File: myhpi/core/templates/core/information_page.html (hunk 4 above)
Comment (abc013) [reply myhpi/core/templates/core/information_page.html:20]: It isn't. However, the label wouldn't be displayed consistently on top then. I guess i could deduplicate it by extracting the code into yet another html file. Would that be better?

Comment (jeriox) [reply myhpi/core/templates/core/information_page.html:20]: No I think that only makes it more complicated. Either leave it this way or put it in the sidebar.html and we accept that it doesn't always show on top. I vote for the latter

File: myhpi/core/templates/core/minutes.html (hunk 6 above)
Comment (abc013) [reply myhpi/core/templates/core/minutes.html]: Hm, according to the page
> A boolean that is set to True when the page is either in draft or published with draft changes.
this also shows if the page is published, but there's a drafted version. I just checked and it indeed also shows the label on the published page if there is a draft.

Comment (abc013) [reply myhpi/core/templates/core/minutes.html]: `page.live` indeed only shows the label if there is _no_ other published version of the page...

File: myhpi/core/templates/core/information_page.html (hunk 4 above)
Comment (abc013) [reply myhpi/core/templates/core/information_page.html]: Done.

File: myhpi/core/templates/core/minutes.html (hunk 6 above)
Comment (jeriox) [reply myhpi/core/templates/core/minutes.html]: but isn't having unpublished changes exactly what we want to protect against? I remember that we had that one minutes that was published nearly empty (but visible for noone) and all of the meaningful content was in a draft

Comment (abc013) [reply myhpi/core/templates/core/minutes.html]: Hm. In this case, we would also need to check the user's permission to change the page (since otherwise, the label would be displayed to everyone). I'll look into that soon.

Comment (jeriox) [reply myhpi/core/templates/core/minutes.html]: I think you are doing the check already to display the page visibility

Comment (abc013) [reply myhpi/core/templates/core/minutes.html]: Updated.

**Review by jeriox — APPROVED** [review]

**Comment by jeriox** [comment]
grafik

**Comment by abc013** [comment]
> grafik

Should be fixed via this PR

**Review by frcroth — APPROVED** [review]
